npm/init.js

```diff
@@ -16,15 +17,30 @@ function validateNpmVersion() {
     }
 }
 
+function download(url) {
+    // We detect proxy by looking at the environment variable
+    var agent = https;
+    if(process.env.https_proxy && process.env.https_proxy.length > 0) {
+        agent = http;
+        url = process.env.https_proxy + starturl.replace('https://', '/https/');
```

Comment:
Can you describe this line?
What is the reason for appending the Bintrat URL to the proxy URL?
Shouldn't the Bintray redirection be configured in the proxy server?

Reply:
This way of building the url with https_proxy environment variable is a convention that is used by many applications like curl, wget, jenkins and many more. I decided will make it a bit more fool proof so that it will work in more proxy scenario's.

Reply:
I will have a good look at this to see if can think of a better solution.

Reply:
OK. The code for downloading over https proxy is now using a connect tunnel just like curl would do when using --proxy parameter. This way it is no longer needed to manipulate the url and is more according to the standards. 

Additional Info: https://github.com/nodejs/node-v0.x-archive/issues/2474

Hoping you will approve. kind regards, Roland